src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1122,38 +1125,207 @@ class spell_pri_power_word_shield : public SpellScriptLoader
+        if (!_spellInfoHeal)
+            return false;
+
+        _healEffectDummy = _spellInfoHeal->GetEffect(EFFECT_0);
+        if (!_healEffectDummy)
+            return false;
+
+        return true;
+    }
+
+    void CastPrayerOfMendingAura(Unit* caster, Unit* target, uint8 stack)
+    {
+        uint32 basePoints = caster->SpellHealingBonusDone(target, _spellInfoHeal, _healEffectDummy->CalcValue(caster), HEAL, _healEffectDummy);
+        CustomSpellValues values;
+        values.AddSpellMod(SPELLVALUE_AURA_STACK, stack);
+        values.AddSpellMod(SPELLVALUE_BASE_POINT0, basePoints);
+        caster->CastCustomSpell(SPELL_PRIEST_PRAYER_OF_MENDING_AURA, values, target, TRIGGERED_FULL_MASK);
+    }
+
+protected:
+    SpellInfo const* _spellInfoHeal;
+    SpellEffectInfo const* _healEffectDummy;
+};
+
+// 33076 - Prayer of Mending
+class spell_pri_prayer_of_mending : public SpellScriptLoader
+{
+public:
+    spell_pri_prayer_of_mending() : SpellScriptLoader("spell_pri_prayer_of_mending") { }
+
+    class spell_pri_prayer_of_mending_SpellScript : public spell_pri_prayer_of_mending_SpellScriptBase
+    {
+        PrepareSpellScript(spell_pri_prayer_of_mending_SpellScript);
+
+        bool Validate(SpellInfo const* spellInfo) override
+        {
+            return spell_pri_prayer_of_mending_SpellScriptBase::Validate(spellInfo) &&
+                ValidateSpellInfo({ SPELL_PRIEST_PRAYER_OF_MENDING_AURA });
+        }
+
+        void HandleEffectDummy(SpellEffIndex /*effIndex*/)
+        {
+            Unit* target = GetExplTargetUnit();
+            if (!target)
+                return;
+
+            CastPrayerOfMendingAura(GetCaster(), target, GetEffectValue());
+        }
+
+        void Register() override
+        {
+            OnEffectHit += SpellEffectFn(spell_pri_prayer_of_mending_SpellScript::HandleEffectDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
+        }
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_pri_prayer_of_mending_SpellScript();
+    }
+};
+
+// 41635 - Prayer of Mending (Aura) - SPELL_PRIEST_PRAYER_OF_MENDING_AURA
+class spell_pri_prayer_of_mending_aura : public SpellScriptLoader
+{
+    public:
+        spell_pri_prayer_of_mending_aura() : SpellScriptLoader("spell_pri_prayer_of_mending_aura") { }
+
+        class spell_pri_prayer_of_mending_aura_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_pri_prayer_of_mending_aura_AuraScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                return ValidateSpellInfo({ SPELL_PRIEST_PRAYER_OF_MENDING_HEAL });
+            }
+
+            void HandleHeal(ProcEventInfo & /*eventInfo*/)
+            {
+                // Caster: the player(priest) that cast the prayer of mending
+                // Target: the player that currently has the prayer of mending aura on him
```

Comment:
Cosmetic/readability:
```suggestion
                // Caster: player(priest) that cast Prayer of Mending
                // Target: player currently wearing Prayer of Mending aura
```